Automattic/wp-calypso
# Jetpack Search: fix enabled check for Jetpack sites

#### Changes proposed in this Pull Request

We currently only look at the site setting `jetpack_search_enabled` to decide whether a site has Jetpack Search enabled in the `/jetpack-search` section. For Jetpack sites, we actually need to check whether the 'search' module is enabled instead.

#### Testing instructions
1. Visit http://calypso.localhost:3000/jetpack-search with a Jetpack site that has Jetpack Search enabled.
2. Ensure that you see the 'enabled' screen:

3. Turn off Jetpack Search in your Settings > Performance.
4. Ensure that you see the 'disabled' screen.

<img width="752" alt="Screen Shot 2021-06-30 at 17 35 04" src="https://user-images.githubusercontent.com/17325/123907274-c44d8280-d9c9-11eb-8c54-1f12ba650e0d.png">

Repeat the process with:
* A WordPress.com simple site
* An Atomic site
* All of the above without a Jetpack Search plan, and make sure you are shown the upsell message instead.

Part of https://github.com/Automattic/wp-calypso/issues/53689. Prior art in https://github.com/Automattic/wp-calypso/pull/53067.
